test/export/InstanceExporter.test.ts

```diff
@@ -766,6 +766,91 @@ describe('InstanceExporter', () => {
+          valueString: 'simple string'
+        }
+      ]);
+    });
+
+    it('should set an extension on meta.profile when a rule sets the InstanceOf url on meta.profile', () => {
+      // Extension: SimpleExt
+      const simpleExt = new Extension('SimpleExt');
+      doc.extensions.set(simpleExt.name, simpleExt);
+      // * meta.profile = "http://hl7.org/fhir/us/minimal/StructureDefinition/TestPatient"
+      // * meta.profile.extension[SimpleExt].valueString = "simple string"
+      const metaProfileValue = new AssignmentRule('meta.profile');
+      metaProfileValue.value = 'http://hl7.org/fhir/us/minimal/StructureDefinition/TestPatient';
+      const metaProfileExtension = new AssignmentRule(
+        'meta.profile.extension[SimpleExt].valueString'
+      );
+      metaProfileExtension.value = 'simple string';
+      patientInstance.rules.push(metaProfileValue, metaProfileExtension);
+      const exported = exportInstance(patientInstance);
+      // since meta.profile already contains a value of instanceOf, nothing changes in either array.
+      expect(exported.meta.profile).toEqual([
+        'http://hl7.org/fhir/us/minimal/StructureDefinition/TestPatient'
+      ]);
+      expect(exported.meta._profile[0].extension).toEqual([
+        {
+          url: 'http://hl7.org/fhir/us/minimal/StructureDefinition/SimpleExt',
+          valueString: 'simple string'
+        }
+      ]);
+    });
+
+    it('should set an extension on meta.profile when a rule sets a non-InstanceOf url on meta.profile', () => {
+      // Extension: SimpleExt
+      const simpleExt = new Extension('SimpleExt');
+      doc.extensions.set(simpleExt.name, simpleExt);
+      // * meta.profile = "http://hl7.org/fhir/us/minimal/StructureDefinition/DifferentPatient"
+      // * meta.profile.extension[SimpleExt].valueString = "simple string"
+      const metaProfileValue = new AssignmentRule('meta.profile');
+      metaProfileValue.value =
+        'http://hl7.org/fhir/us/minimal/StructureDefinition/DifferentPatient';
+      const metaProfileExtension = new AssignmentRule(
+        'meta.profile.extension[SimpleExt].valueString'
+      );
+      metaProfileExtension.value = 'simple string';
+      patientInstance.rules.push(metaProfileValue, metaProfileExtension);
+      const exported = exportInstance(patientInstance);
+      // in this case, since profile[0] has both a value and a child element before receiving instanceOf,
+      // keep those aligned.
+      expect(exported.meta.profile).toEqual([
+        'http://hl7.org/fhir/us/minimal/StructureDefinition/TestPatient',
+        'http://hl7.org/fhir/us/minimal/StructureDefinition/DifferentPatient'
+      ]);
+      expect(exported.meta._profile).toEqual([
+        null,
+        {
+          extension: [
+            {
+              url: 'http://hl7.org/fhir/us/minimal/StructureDefinition/SimpleExt',
+              valueString: 'simple string'
+            }
+          ]
+        }
+      ]);
+    });
+
```

Comment:
Can we add two more tests?
1. Set a different canonical (and extension) directly on `meta.profile[1]` (leaving `meta.profile[0]` empty for SUSHI to put in the default.
2. Set a different canonical at `meta.profile[0]` (w/ the extension) and the instanceOf canonical at `meta.profile[1]` (also w/ the extension). We want to be sure that the depuplication code is working as expected (spoiler alert: I'm pretty sure it's not, see [InstanceExporter.ts#839](https://github.com/FHIR/sushi/blob/e10d628e2534bd07df1b8894c847d1eaec0900be/src/export/InstanceExporter.ts#L839)).